src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -500,11 +509,15 @@ private void QueueApplyReplacements()
                     TaskContinuationOptions.OnlyOnRanToCompletion,
                     TaskScheduler.Default)
                 .Unwrap()
-                .SafeContinueWith(
-                    t => ApplyReplacements(t.Result.replacementInfo, t.Result.mergeResult, _conflictResolutionTaskCancellationSource.Token),
+                .SafeContinueWithFromAsync(
+                    async t =>
+                    {
+                        await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_conflictResolutionTaskCancellationSource.Token);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.